src/deferred.ts

```diff
@@ -11,7 +11,12 @@ const DEBUG = false;
 
 let id = 0;
 
-class Deferred {
+export class Deferred {
+  private id: number;
+  public promise: Promise<any>;
```

Comment:
The promise isn't used here so I assumed it should be public.

Reply:
Yes, it should be public.